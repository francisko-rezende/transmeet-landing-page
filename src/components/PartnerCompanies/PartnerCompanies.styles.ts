import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  margin-top: 22.3rem;
  display: grid;
  justify-content: center;
`
export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.grayLight};
    font-size: ${theme.font.sizes['4']};
    font-weight: ${theme.font.weight.regular};
    text-align: center;
  `}
`

export const IconsWrapper = styled.div`
  margin-top: 4.1rem;
  display: flex;
  align-items: center;
  gap: 60px;
`
