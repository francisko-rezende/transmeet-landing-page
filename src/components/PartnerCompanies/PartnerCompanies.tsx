import React from 'react'

import MainContainer from 'components/MainContainer'
import Image from 'next/image'

import * as S from './PartnerCompanies.styles'

const PartnerCompanies = () => {
  return (
    <MainContainer>
      <S.Wrapper>
        <S.Title>- conheça alguns dos nossos parceiros -</S.Title>
        <S.IconsWrapper>
          <Image
            src="/images/alpargatas-icon.webp"
            width={120}
            height={61}
            alt="alpagartas"
          />
          <Image
            src="/images/google-icon.webp"
            width={120}
            height={39}
            alt="google"
          />
          <Image
            src="/images/pitagoras-icon.webp"
            width={80}
            height={58}
            alt="pitágoras"
          />
          <Image
            src="/images/raizen-icon.webp"
            width={120}
            height={38}
            alt="raízen"
          />
          <Image
            src="/images/santander-icon.webp"
            width={120}
            height={22}
            alt="santander"
          />
          <Image
            src="/images/ioasys-icon.webp"
            width={80}
            height={60}
            alt="ioasys"
          />
        </S.IconsWrapper>
      </S.Wrapper>
    </MainContainer>
  )
}

export default PartnerCompanies
