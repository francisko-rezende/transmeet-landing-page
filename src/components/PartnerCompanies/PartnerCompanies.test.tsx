import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import PartnerCompanies from '.'

describe('<PartnerCompanies />', () => {
  test('should render the partner logos correctly', () => {
    const { container } = renderWithTheme(<PartnerCompanies />)

    expect(
      screen.getByRole('heading', {
        name: /- conheça alguns dos nossos parceiros -/i
      })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /alpagartas/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /google/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /pitágoras/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /raízen/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /santander/i })).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /ioasys/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
