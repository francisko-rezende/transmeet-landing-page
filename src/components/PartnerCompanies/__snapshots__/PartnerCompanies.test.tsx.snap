// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartnerCompanies /> should render the partner logos correctly 1`] = `
.c0 {
  max-width: calc(1170px + 44px);
  padding: 0 22px;
  margin: 0 auto;
}

.c1 {
  margin-top: 22.3rem;
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  color: #A0A4A7;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
}

.c3 {
  margin-top: 4.1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 60px;
}

@media (max-width:811px) {
  .c1 {
    margin-top: 5rem;
  }
}

@media (max-width:610px) {
  .c2 {
    font-size: 1rem;
    font-weight: 700;
  }
}

@media (max-width:811px) {
  .c3 {
    gap: 40px;
  }
}

@media (max-width:400px) {
  .c3 {
    margin-top: 30px;
    gap: 14px;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <h2
      class="c2"
    >
      - conheça alguns dos nossos parceiros -
    </h2>
    <div
      class="c3"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2761%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="alpagartas"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2739%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="google"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2758%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="pitágoras"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2738%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="raízen"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2722%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="santander"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2760%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="ioasys"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </section>
</div>
`
